plugins/source-kafka/pkg/client/client.go

```diff
@@ -64,7 +61,11 @@ func (c *kafkaSourceClient) KafkaSourceParams() *types.KafkaSourceParams {
 
 //CreateKafkaSource is used to create an instance of KafkaSource
 func (c *kafkaSourceClient) CreateKafkaSource(kafkaSource *v1alpha1.KafkaSource) error {
-	_, err := c.client.KafkaSources(c.namespace).Create(kafkaSource)
+	sourceClient, err := c.KafkaClient()
+	if err != nil {
+		return knerrors.GetError(err)
+	}
```

Comment:
I think we should raise this error in `NewKafkaSourceClient` function itself while creating the client.
So, `NewKafkaSourceClient` handles any error for creating the client while `CreateKafkaSource` could be focused on any error raised while creating source itself.

This will also prevent repetition of this error handling in each CREATE/GET/DELETE.

Reply:
fixed. Please review again.